extension/src/providers/gotoProvider.ts

```diff
@@ -0,0 +1,145 @@
+					locations.push(item);
+				});
+			} else if (parentContainer) {
+				this.findFirstVariableMatch(rDoc, position, selected, parentContainer).forEach(item => {
+					locations.push(item);
+				});
+			}
+			return locations;
+		} catch (error) {
+			return;	// I don't believe this requires a localized error since VSCode has a default "no definition found" response
+		}
+	}
+
+	
+	private findFirstVariableMatch(doc: ReadonlyDocument, start: vscode.Position, searchFor: string, searchIn: Sexpression): vscode.Location[] {
+		const result: vscode.Location[] = [];
+		const ucName = searchFor.toUpperCase();
+		let context = searchIn.getSexpressionFromPos(start);
+		let flag = true;
+		do {
+			const parent = searchIn.getParentOfSexpression(context);
+			const atom = parent?.getNthKeyAtom(0);
+			if (atom && /^DEFUN$|^DEFUN-Q|^LAMBDA$/i.test(atom.symbol)) {
+				let headers = parent.getNthKeyAtom(1);
+				if (headers?.symbol.toUpperCase() === ucName){ // adds defun names to possible result. Especially useful for quoted function names.
+					result.push(new vscode.Location(vscode.Uri.file(doc.fileName), new vscode.Position(headers.line, headers.column)));
+				}
+				if (!(headers instanceof Sexpression)){
+					headers = parent.getNthKeyAtom(2);
+				}
+				if (headers instanceof Sexpression){
+					const found = headers.atoms.find(p => p.symbol.toUpperCase() === ucName);
+					if (found) {
+						result.push(new vscode.Location(vscode.Uri.file(doc.fileName), new vscode.Position(found.line, found.column)));
+					}
+				}
+			} else if (atom && /^FOREACH$|^VLAX-FOR$/i.test(atom.symbol)) {
+				const tmpVar = parent.getNthKeyAtom(1);
+				if (!(tmpVar instanceof Sexpression) && tmpVar.symbol.toUpperCase() === ucName){
+					result.push(new vscode.Location(vscode.Uri.file(doc.fileName), new vscode.Position(tmpVar.line, tmpVar.column)));
+				}
+			}
+			if (!parent || result.length > 0){
+				flag = false;
+			} else {
+				context = parent;
+			}
+		} while (flag);
+		// If we still haven't found anything check the 1st occurrence of setq's. This will find globals setqs and nested ones possibly inside other defuns
+		if (result.length === 0){
+			doc.findExpressions(/^SETQ$/i).forEach(setq => {
+				let isVar = false;
+				let cIndex = setq.nextKeyIndex(0);
+				do {
+					const atom = setq.atoms[cIndex];
+					if (isVar && atom?.symbol.toUpperCase() === ucName) {
+						result.push(new vscode.Location(vscode.Uri.file(doc.fileName), new vscode.Position(atom.line, atom.column)));
+					}
+					cIndex = setq.nextKeyIndex(cIndex);
+					isVar = !isVar;
+				} while (cIndex && cIndex > -1);
+			});
+		}
+		return result;
+	}
+
+	private findDefunMatches(searchFor: string, searchIn: ReadonlyDocument[]): vscode.Location[] {
+		const result: vscode.Location[] = [];
+		const regx = new RegExp('\\((DEFUN|DEFUN-Q)' + escapeRegExp(searchFor) + '\\(', 'ig');
+		searchIn.forEach((doc: ReadonlyDocument) => {
+			if (regx.test(doc.fileContent.replace(/\s/g, ''))){
```

Comment:
what is the \s/g

Reply:
I am attempting to evaluate this without the context of tabs, line feeds and spaces. Basically if someone decided to they like this format declaration it won't break the primitive search:

```lisp
(defun
       whatever 
       (a b / c d e)
  (princ (setq c (+ a b)))
)
```